contributing.md

```diff
@@ -90,10 +90,18 @@ This will create an `amp.zip` in the plugin directory which you can install. The
 ## Updating Allowed Tags And Attributes
 
 The file `class-amp-allowed-tags-generated.php` has the AMP specification's allowed tags and attributes. It's used in sanitization.
-To update that file:
+
+To update that file, perform the following steps (which need to be improved):
```

Comment:
We could automate the fetching of the latest version by getting `ampRuntimeVersion` from https://cdn.ampproject.org/rtv/metadata while chopping off the `01` release prefix.

But still, the method of including `ampproject/amphtml` repo as a Composer dependency seems hacky here. There are 4 tag references that have to be maintained. @swissspidy any thoughts on a better way to include the dependency?

Reply:
Is there any benefit of having that repo as a Composer dependency, except for convenience and perhaps visibility of what version we're using?

Hmm on first thought, we could do this:

1. Remove dependency from `composer.json`
2. Extend `bin/amphtml-update.sh`:  
   1. Fetch version number from https://cdn.ampproject.org/rtv/metadata
   2. Download the respective ZIP file from GitHub to a temporary location
   3. Run python script
   4. Delete temporary folder again
3. Bonus: Add `AMP__AMPHTML_VERSION` constant to `amp.php` and update that with the new version number each time.
3. Stretch goal: run that script on Travis CI and fail builds (or throw warnings or something) if the current spec isn't up to date.
4. Stretch goal: have a bot run that script and open pull requests

Reply:
I suppose the main benefit of having the Composer dependency is that the plugin gets explicitly tied to a version of AMP. In the past it has been annoying to keep track of the current version of AMP that is applied, but I suppose this would be minimized by a `AMP__AMPHTML_VERSION` constant…

> 3\. Bonus: Add `AMP__AMPHTML_VERSION` constant to `amp.php` and update that with the new version number each time.

There is some of this currently in:

https://github.com/ampproject/amp-wp/blob/c9e21c766a2bf84ac48c88dbe03391abf7b62b59/includes/sanitizers/class-amp-allowed-tags-generated.php#L16-L17

But it doesn't indicate the actual AMP version. Also, these private static variables aren't even used anywhere. So the version could be added to that class instead as a const.

Reply:
I'll implement those changes.

Reply:
Awesome! Let me know if I can help with anything.

Reply:
OK, I've now implemented this. However, I opted to not do the `AMP__AMPHTML_VERSION` constant at this time because it would not enable us to try generating the spec from a non-tagged version. 

As it stands right now, you can have a local clone of amphtml and generate the spec to whatever commit or branch you have checked out there by doing:

```basu
./bin/amphtml-update.sh amphtml/
```

Where `amphtml` is the Git repo. If you omit that, however, then it gets the current stable release.

We can revisit how we incorporate the amphtml version into the plugin later. Right now it's not strictly needed.

Reply:
Note that automating the update via Travis would not probably be worth it because the spec changes from time to time and changes are needed for the Python script to correctly generate from the spec. Also, it's important to also add unit tests to make sure that new tags/attributes are handled properly. So I think it should be a manual process when new versions come out for the forseeable future.